lib/mauth/client/security_token_cacher.rb

```diff
@@ -61,6 +54,7 @@ def signed_mauth_connection
           @signed_mauth_connection ||= ::Faraday.new(@mauth_client.mauth_baseurl, @mauth_client.faraday_options) do |builder|
             builder.use MAuth::Faraday::MAuthClientUserAgent
             builder.use MAuth::Faraday::RequestSigner, 'mauth_client' => @mauth_client
+            builder.use Faraday::HttpCache, store: Rails.cache, serializer: Oj, logger: Rails.logger
```

Comment:
Isn't this much heavier than before? Before we were taking the key directly from memory, now we need to pretend we are going to do an http request, call an external cache system, deserialize the thing and get the object

Reply:
i mean we could also do an in-memory store, which i think is the default when u give no `store` option here:
https://github.com/sourcelevel/faraday-http-cache

obv there is a tradeoff in that the caches will then be specific to the memory of each running instance as opposed to shared across all instances via redis.  not sure which would be preferable.

Reply:
I believe MAuth returns a response with `Cache-Control: private`:
https://github.com/mdsol/mauth/blob/release.2019.2.1/app/controllers/security_tokens_controller.rb#L14

Can you add `shared_cache: false`?
https://github.com/sourcelevel/faraday-http-cache#shared-vs-non-shared-caches

Reply:
Thanks @ykitamura-mdsol That's useful